File: problems/All Paths From Source to Target/allPathsSourceTarget.cs
public class Solution {
    public IList<IList<int>> AllPathsSourceTarget(int[][] graph) {
        var result = new List<IList<int>>();

        dfs(graph, new List<int>(), result);

        return result;
    }

    private void dfs(int[][] graph, List<int> path, IList<IList<int>> result, int idx = 0) {
        path.Add(idx);

        if (graph.Length - 1 == idx) {
            result.Add(path);
        } else {
            foreach (var item in graph[idx]) {
                dfs(graph, new List<int> (path), result, item);
            }
        }
    }
}

File: problems/All Paths From Source to Target/allPathsSourceTarget.js
/**
 * @param {number[][]} graph
 * @return {number[][]}
 */
var allPathsSourceTarget = function(graph) {
    const result = [];

    dfs(graph, [], result, 0);

    return result;
};

const dfs = function(graph, path, result, idx) {
    path.push(idx);

    if (graph.length == 1 + idx) {
        result.push(path);
    } else {
        for (const child of graph[idx]) {
            dfs(graph, [...path], result, child);
        }
    }
}
